test/functional/update_v2/update-delta-manifest.bats

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bats
+
+load "../testlib"
+
+test_setup() {
+
+	create_test_environment "$TEST_NAME" 10
+	create_bundle -L -n bundle1 -v 10 -f /file1 "$TEST_NAME"
+	create_bundle -L -n bundle2 -v 10 -f /file2 "$TEST_NAME"
+
+	create_version "$TEST_NAME" 20 10 staging
+	update_bundle -i "$TEST_NAME" bundle1 --update /file1
+
+	create_version "$TEST_NAME" 30 20 staging
+	update_bundle -i "$TEST_NAME" bundle2 --update /file2
+}
+
+@test "update with delta manifests" {
+
+	run sudo sh -c "$SWUPD update $SWUPD_OPTS"
+	assert_status_is 0
+
+	# Verify target files
+	assert_file_exists "$TARGETDIR"/file1
+	assert_file_exists "$TARGETDIR"/file2
+
+	# Verify delta manifests cleaned from state dir
+	assert_file_not_exists "$STATEDIR"/20/Manifest.bundle1.D.10
+	assert_file_not_exists "$STATEDIR"/30/Manifest.bundle2.D.20
+
+	# Verify unnecessary manifests not downloaded
+	assert_file_not_exists "$STATEDIR"/10/Manifest.bundle1
+	assert_file_not_exists "$STATEDIR"/10/Manifest.bundle2
+
+	expected_output=$(cat <<-EOM
+		Update started.
+		Preparing to update from 10 to 30
+		Downloading packs...
+		Extracting bundle1 pack for version 20
+		Extracting bundle2 pack for version 30
+		Statistics for going from version 10 to version 30:
+		    changed bundles   : 2
+		    new bundles       : 0
+		    deleted bundles   : 0
+		    changed files     : 2
+		    new files         : 0
+		    deleted files     : 0
+		Starting download of remaining update content. This may take a while...
+		Finishing download of update content...
+		Staging file content
+		Applying update
+		Update was applied.
+		Calling post-update helper scripts.
+		Update successful. System updated from version 10 to version 30
+	EOM
+	)
+	assert_in_output "$expected_output"
+}
```

Comment:
What about the case when a bundle has updates during a couple of versions?
For example:

create_test_environment "$TEST_NAME" 10
create_bundle -L -n bundle1 -v 10 -f /file1 "$TEST_NAME"
create_version "$TEST_NAME" 20 10 staging
update_bundle -i "$TEST_NAME" bundle1 --update /file1
create_version "$TEST_NAME" 30 20 staging
update_bundle -i "$TEST_NAME" bundle1 --update /file1

So if the current version is 10 and we do an update, it should update all the way to version 30, iterative and delta manifests are going to exist for each version, but since the jump is larger than one version the delta manifests should not be used, correct? (I know the iterative manifest should not be used)
Maybe a test for this scenario is worth adding.

Reply:
My understanding is that the server would look like the following:

* 20
    * pack-bundle1-from-0.tar
    * pack-bundle1-from-10.tar
* 30
    * pack-bundle1-from-0.tar
    * pack-bundle1-from-10.tar
    * pack-bundle1-from-20.tar

When updating FROM 10 TO 30, a valid delta manifest would be extracted from /30/pack-bundle1-from-10.tar. This delta manifest should include entries for the files that changed during the update FROM 10 TO 30.

Reply:
@otaviobp Is my assumption wrong here? Are delta manifests only valid for the N-1 version?

If delta manifests only work for the N-1 verison, I will need to not look for old delta manifests.

Reply:
Yes @jwakre right now we are only publishing delta manifests for bundle->previous (not necessarily N-1 or N-10)

Reply:
Wait, no that's wrong. Delta manifests are valid for the FROM version to the TO version. I was getting my iterative and delta manifests mixed up again (I've been doing that a lot). If you download a pack from FROM (to your TO) it will have all the records of files used in deltas.

Reply:
tl;dr @jwakre is right

Reply:
Ok, thanks for the clarification. I think a test case for the scenario when a bundle has been updated more than once since an update, so I will add that.